### PR TITLE
RELEASE: Update NEWS 1.16 RC4

### PR DESCRIPTION
**Pull request by shasson5**
## What
Update NEWS 1.16 RC4

## Why ?
Prepare v.16 RC4

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -13,8 +13,14 @@
 
 ## 1.16.0-rc4 (February 21, 2024)
 ### Bugfixes:
+#### UCP
+* Disabled rendezvous pipeline protocol selection when using non-contiguous buffer.
 #### UCT
 * Fixed Infiniband posting error when compiler optimizes code with memmove()
+#### GPU (CUDA, ROCM)
+* Fixed CUDA IPC disabling when fails to initialize.
```
Comment (yosefe) [inline NEWS:21]: this is not clear, need to rephrase, what fails to initialize and in which cases?

File: NEWS [hunk 2]
```diff
@@ -13,8 +13,14 @@
...
+#### Build
+* Fixed compilation error when building on PPC64.
```
Comment (yosefe) [inline NEWS:23]: isn't it UCM fix? should be in UCM section

File: NEWS [hunk 3]
```diff
@@ -13,8 +13,14 @@
 
 ## 1.16.0-rc4 (February 21, 2024)
 ### Bugfixes:
+#### UCP
+* Disabled rendezvous pipeline protocol selection when using non-contiguous buffer.
 #### UCT
 * Fixed Infiniband posting error when compiler optimizes code with memmove()
+#### GPU (CUDA, ROCM)
+* Fixed CUDA IPC disabling when CUDA device is not set for the process.
```
Comment (yosefe) [inline NEWS:21]: Fixed cuda_ipc transport being disabled if a CUDA device is not set during initialization

File: NEWS [hunk 4]
```diff
@@ -13,8 +13,14 @@
 
 ## 1.16.0-rc4 (February 21, 2024)
 ### Bugfixes:
+#### UCP
```
Comment (tvegas1) [inline NEWS:16]: can we also add details for #9723?

**Review by yosefe — APPROVED** [review]

File: NEWS (hunk 4 above)
Comment (shasson5) [reply NEWS:16]: isn't it covered by the comment for #9705 ?

**Comment by shasson5** [comment]
@yosefe can merge?

File: NEWS (hunk 4 above)
Comment (yosefe) [reply NEWS:16]: Maybe rephrase it to "Fixed mlx5 WQE posting error due to compiler memory copy optimizations"
and rename the section from "UCT" to "RDMA CORE" like in line 56

**Review by yosefe — APPROVED** [review]

**Comment by shasson5** [comment]
@yosefe squashed
